webapp/store/index.js

```diff
@@ -16,6 +16,24 @@ import {
 } from '../plugins/plugins';
 import * as reducers from './reducers';
 
+function errorCatcherMiddleware(errorHandler) {
+  return function(store) {
+    return function(next) {
+      return function(action) {
```

Comment:
Could you make this a named function so that when debugging, it shows up in the stack trace and is very obvious?

Reply:
I believe this is the named function syntax no? the name should show up in the stack trace https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions#The_function_expression_(function_expression)

Reply:
```
function [name]([param[, param[, ... param]]]) {
   statements
}
```

On line 22 in particular I meant, the stack would show a bunch of anonymous functions before `errorCatcherMiddleware`

Reply:
Ah ok, sorry misunderstood the comment. it looks funny but that's actually just the convention that's used for redux middleware (same as in the [redux docs](https://redux.js.org/advanced/middleware#the-final-approach)). I could use arrow functions to make it more readable though: 
```js
function errorCatcherMiddleware(errorHandler) {
  return store => next => action => {

  }
}
```
which is actually what we do for our plugin middleware already. Good thing is that the error is still very readable (though the anonymous functions do come up in the stack trace as you point out):

<img width="1355" alt="screen shot 2018-08-20 at 5 01 07 pm" src="https://user-images.githubusercontent.com/4344978/44372953-64090900-a49b-11e8-9005-ff0564ca2072.png">